src/actions/AuthActions.js

```diff
@@ -1,67 +1,105 @@
-import firebase from 'firebase';
+import {AsyncStorage} from 'react-native';
+import {Facebook} from 'expo';
+import axios from 'axios';
+
 import _ from 'lodash';
 
 import {
-    EMAIL_CHANGED,
-    PASSWORD_CHANGED,
-    LOGIN_AUTH_ERROR,
-    LOGIN_SUCCESS
+    APP_READY,
+    FACEBOOK_LOGIN_SUCCESS,
+    FACEBOOK_LOGIN_FAILURE,
+    FACEBOOK_LOGOUT_COMPLETE,
+    GUEST_TOKEN_SET
 } from './types';
 
-export const emailChanged = (text) => {
-    return {
-        type: EMAIL_CHANGED,
-        payload: text
-    };
-};
+import {FACEBOOK_AUTH} from '../../env';
+import {tokens} from '../constants';
 
-export const passwordChanged = (text) => {
-    return {
-        type: PASSWORD_CHANGED,
-        payload: text
+const attemptFacebookLogin = async (dispatch) => {
+    const {appID} = FACEBOOK_AUTH;
+    const faceBookOptions = {
+        permissions: ['public_profile']
     };
-};
 
-export const loginAuthError = (dispatch, err) => {
-    dispatch({
-        type: LOGIN_AUTH_ERROR,
-        payload: err
-    });
-};
+    let {type, token} = await Facebook.logInWithReadPermissionsAsync(appID, faceBookOptions);
 
-export const loginSuccess = (dispatch, token) => {
-    // TODO token arg will be used when we add FB Auth
-    let tokenToSend = token;
+    if (type === 'success') {
+        let {data: {email, name, picture: {data: {url}}}} = await axios.get(`https://graph.facebook.com/me?fields=id,name,email,picture&access_token=${token}`);
+        // we now have access to email, name, and their profile picture
 
-    if (!tokenToSend) {
-        tokenToSend = 'Manually Generated Token'
-    }
+        await AsyncStorage.setItem(tokens.USER_TOKEN, token);
 
-    dispatch({
-        type: LOGIN_SUCCESS,
-        payload: tokenToSend
-    });
+        dispatch({
+            type: FACEBOOK_LOGIN_SUCCESS,
+            payload: token
+        })
+    } else {
+        dispatch({
+            type: FACEBOOK_LOGIN_FAILURE
+        })
+    }
 };
 
-export const loginUser = ({email, password}) => {
-    return (dispatch) => {
-        firebase.auth().signInWithEmailAndPassword(_.trim(email), _.trim(password))
-            .then(user => {
-                loginSuccess(dispatch);
-            })
-            .catch(err => {
-                loginAuthError(dispatch, err);
-            })
+export const logUserIntoFacebook = () => {
+
+    return async (dispatch) => {
+        let token = await AsyncStorage.getItem(tokens.USER_TOKEN);
+
+        if (token) {
+            checkAndSetToken(token);
+        } else {
+            attemptFacebookLogin(dispatch);
+        }
```

Comment:
When we click on "Sign in with Facebook", first double check to make sure we don't have a token (in case for some reason we missed it). If we do, then call `checkAndSetToken` (which is shown next), if not then kick off the facebook process with `attemptFacebookLogin` which we saw above.